scripts/deploymentextensions/fixlowstorage/sharedOperations.sh

```diff
@@ -0,0 +1,185 @@
+        true
+    fi
+
+    # else case: the non-zero (failing) exit code will bubble up to caller.
+}
+
+source_utilities()
+{
+    # expected case
+    source_wrapper "utilities.sh" && return
+
+    # created
+    actual_utilities_path=$current_script_path/../../../templates/stamp/utilities.sh
+    local_utilities_path=$current_script_path/utilities.sh
+    ln -s $actual_utilities_path $local_utilities_path
+
+    # expected case
+    source_wrapper "utilities.sh"
+}
+
+parse_args()
+{
+    while [[ "$#" -gt 0 ]] ; do
+        arg_value="${2}"
+        shift_once=0
+
+        if [[ "${arg_value}" =~ "--" ]] ; then
+            arg_value=""
+            shift_once=1
+        fi
+
+         # Log input parameters to facilitate troubleshooting
+        log "Option '${1}' set with value '"${arg_value}"'"
+
+        case "$1" in
+          --script-file)
+            script_file="${arg_value}"
+            ;;
+          --settings-file)
+            settings_file="${arg_value}"
+            ;;
+          --backend-server-list)
+            backend_server_list=(`echo ${arg_value} | base64 --decode`)
+            ;;
+          --target-user)
+            target_user="${arg_value}"
+            ;;
+          --paths-to-copy-list)
+            paths_to_copy_list=(`echo ${arg_value} | base64 --decode`)
+            ;;
+          --destination-path)
+            destination_path="${arg_value}"
+            ;;
+          --remote-command)
+            remote_command="${arg_value}"
+            ;;
+          --low-storage-log)
+            low_storage_log="${arg_value}"
+            ;;
+          --low-storage-frequency)
+            low_storage_frequency="${arg_value}"
+            ;;
+          --usage-threshold-percent)
+            usage_threshold_percent="${arg_value}"
+            ;;
+          --file-size-threshold)
+            file_size_threshold="${arg_value}"
+            ;;
+          --mysql-user)
+            mysql_user="${arg_value}"
+            ;;
+          --mysql-pass)
+            mysql_pass="${arg_value}"
+            ;;
+          --large-partition)
+            large_partition="${arg_value}"
+            ;;
+        esac
+
+        shift # past argument or value
+
+        if [[ $shift_once -eq 0 ]] ; then
+            shift # past argument or value
+        fi
+
+    done
+}
+
+###############################################
+# INVOKED EXTERNALLY
+###############################################
+persist_settings_for_cron()
+{
+# persist the settings
+    bash -c "cat <<EOF >${settings_file}
+backend_server_list=\"$backend_server_list\"
+target_user=$target_user
+paths_to_copy_list=\"$paths_to_copy_list\"
+destination_path=$destination_path
+remote_command=$remote_command
+remote_arguments=\"$@\"
+EOF"
+
+    # this file contains important information (like db info). Secure it
+    chmod 600 $settings_file
+}
+create_or_update_cron_job()
+{
+    # create the cron job
+    cron_installer_script="${script_file}.cron"
```

Comment:
- [ ] what happens if $script_file is blank / not specified?
- [ ] make sure script_file has an absolute path reference